lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -176,13 +176,40 @@ def credit(amount, payment, options = {})
         end
       end
 
-      def verify(credit_card, options = {})
+      def verify(payment_method, options = {})
+        amount = define_amount_for_verify(payment_method, options)
+        return amount if amount.class == Response
+
         MultiResponse.run(:use_first_response) do |r|
-          r.process { authorize(100, credit_card, options) }
-          r.process(:ignore_result) { void(r.authorization, options) }
+          r.process { authorize(amount, payment_method, options) }
+          r.process(:ignore_result) { void(r.authorization, options) } unless amount == 0
         end
       end
 
+      def define_amount_for_verify(payment_method, options)
+        if options[:verify_amount].present?
+          if options[:verify_amount].class != Integer || options[:verify_amount] < 0
+            response = Response.new(false, 'verify_amount value must be an integer')
```

Comment:
I have checked this and need to comment we tried this because otherwise we will not be able to support the default flow, additionally, about the response, this was the way for us to raise the error in a proper way to have the possibility to test it in the remote tests.
If you have any other suggestion or recommendation about this I will be happy to check it

Reply:
I see, thank you for the explanation.